src/models/train.py

```diff
@@ -90,7 +92,7 @@ def instantiate_trainer(self, **kwargs: Any) -> Trainer:
 
         if self.datamodule.val_path:
             if self.datamodule.is_classification:
-                checkpoint_metric = "val/roc_auc"
+                checkpoint_metric = "val/{}".format(self.config["fit"]["optimize"])
                 mode = "max"
```

Comment:
Would it be hard to add a second argument for `min/max`? For example, this would break if we tried to minimize `train/loss` for a regression task. Maybe one solution is to have the `optimize` argument take two strings (a metric and `"min"|"max"`, like this: https://stackoverflow.com/a/16968580